.github/workflows/pull_request.yml

```diff
@@ -94,3 +94,49 @@ jobs:
 
       - name: Run universe tests
         run: ./mvnw -pl :universe verify -Puniverse -Dmaven.javadoc.skip=true
+
+  compatibility_test:
+    # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on
+    runs-on: ubuntu-18.04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cancel Previous Runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - uses: satackey/action-docker-layer-caching@v0.0.8
+        continue-on-error: true
+
+      - name: Prepare top of trunk docker image
+        run: ./mvnw clean install -DskipTests -Pcompatibility -Dmaven.javadoc.skip=true -T 1C
+
+      - name: Pull docker image built from master parent commit
+        run: docker pull corfudb/corfu-server:0.3.1-SNAPSHOT
```

Comment:
It seems to me you don't need to do docker pull, universe doing it actomatically

Reply:
yeah -Pdocker will build a docker image  with `0.3.1-SNAPSHOT` tagged, but that will be the top of trunk version. Here we want `0.3.1-SNAPSHOT` to be the version which is already hosted in docker hub, i.e. the parent version.
But yeah this is a little confusing, we may come up with better versioning strategy later..